modules/core/src/main/scala/Span.scala

```diff
@@ -22,3 +22,18 @@ trait Span[F[_]] {
   def span(name: String): Resource[F, Span[F]]
 
 }
+
+
+sealed trait Propagation extends Product with Serializable
+object Propagation {
+
+  /** Fields will not be propagated to child spans. */
+  case object None     extends Propagation
+
+  /** Fields will be propagated to child spans, but will not cross network boundaries. */
```

Comment:
Is there a case for propagating Local fields back out to parent spans as well as into child spans? For example, I have a situation where I'm inserting an entity into the db, allocating a primary key at that point, and I'm adding the generated key to the span that wraps the db call. But it would be valuable if that key also appeared on the parent spans so that I can easily cross-reference them.

I can see, though, that there would be cases to address such as what to do if the parent span already has a value for a Local field set by a child span.